master/internal/api_task.go

```diff
@@ -0,0 +1,27 @@
+package internal
+
+import (
+	"context"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/pkg/errors"
+
+	"github.com/determined-ai/determined/master/internal/db"
+	"github.com/determined-ai/determined/proto/pkg/apiv1"
+	"github.com/determined-ai/determined/proto/pkg/taskv1"
+)
+
+func (a *apiServer) GetTask(
+	_ context.Context, req *apiv1.GetTaskRequest) (resp *apiv1.GetTaskResponse, err error) {
+	t := &taskv1.Task{}
+	switch err := a.m.db.QueryProto("get_task", t, req.TaskId); err {
+	case db.ErrNotFound:
+		return nil, status.Errorf(
+			codes.NotFound, "task %s not found", req.TaskId)
+	default:
+		return &apiv1.GetTaskResponse{Task: t},
+			errors.Wrapf(err, "error fetching task %s from database", req.TaskId)
```

Comment:
**question:** I looked at other endpoint examples that do this, but figure you can just tell me. If `err` ends being something other than `db.ErrNotFound` (such as `db. ErrUnsupported`), it'll reach the `default` case and then would it return  `errors.Wrapf(...)` instead of the `{ Task: t }` response?

Reply:
yes that sounds right, it's based on GetModel and similar endpoints

Reply:
yeah, personally i dont love this, but it works. it relies on errors.Wrap returning nil if the err is nil and it being ok to return a response and err in the err case